src/exec.cpp

```diff
@@ -998,7 +1002,19 @@ void Executor::raise(const std::string &exception, const std::string &info)
         ip = callstack.top();
         callstack.pop();
     }
-    fprintf(stderr, "unhandled exception %s (%s)\n", exception.c_str(), info.c_str());
+    fprintf(stderr, "unhandled exception %s (%s) at address %lu\n", exception.c_str(), info.c_str(), eip);
+
+    if (debug != nullptr) {
+        auto p = eip;
+        auto line = debug->line_numbers.end();
+        while (line == debug->line_numbers.end()) {
+            line = debug->line_numbers.find(p--);
```

Comment:
Don't use the inline decrement operator. I find this really hard to read and to think about whether it's correct or not.